scripts/prepareForFirstWeekLaunch.ts

```diff
@@ -33,13 +36,95 @@ async function main() {
   const sdl = (await ethers.getContract("SDL")) as SDL
   const veSDL = (await ethers.getContract("VotingEscrow")) as VotingEscrow
   const minter = (await ethers.getContract("Minter")) as Minter
-  const DAY = 86400
   const WEEK = 86400 * 7
   const YEAR = WEEK * 52
-  const GaugeHelperContract = (await ethers.getContract(
+  const gaugeHelperContract = (await ethers.getContract(
     "GaugeHelperContract",
   )) as GaugeHelperContract
-  //  Deploy dummy reward tokens
+
+  const numberOfGauges = (await gaugeController.n_gauges()).toNumber()
+  const gauges: { name: string; address: string }[] = []
+
+  for (let i = 0; i < numberOfGauges; i++) {
+    const address = await gaugeController.gauges(i)
+    const gaugeContract = (await ethers.getContractAt(
+      "LiquidityGaugeV5",
+      address,
+    )) as LiquidityGaugeV5
+
+    let name: string
+    try {
+      name = await gaugeContract.name()
+    } catch (e) {
+      // In case of root gauges, they don't have a name.
+      name = address
+    }
+
+    gauges.push({ name, address })
+  }
+
+  // The deploy scripts should have already added a default gauge type (should not include root gauges for)
+  expect((await gaugeController.n_gauge_types()).toNumber()).to.eq(1)
+  expect((await gaugeController.gauge_type_names(0)).toString()).to.eq(
+    "Liquidity",
+  )
+
+  // Test calling gaugeHelperContract that reads in series
+  console.log(
+    `${gauges[3].name} address: `,
+    await gaugeHelperContract.gaugeToPoolAddress(gauges[3].address),
+  )
+  console.log(await gaugeHelperContract.gaugeToPoolData(gauges[3].address))
+
+  // You can freely modify timestamps and the state of the contracts to your liking.
+  // For how you want to set up the contracts, please refer to test files in test/tokenomics
+
+  // Ensure sdl is not paused
+  if (await sdl.paused()) {
+    await sdl.enableTransfer()
+  }
+
+  // Only manually trigger mining parameters if rate is 0 (uninitialized)
+  const rewardRate = await minter.rate()
+  if (rewardRate.eq(0)) {
+    await minter.update_mining_parameters()
+  }
+
+  // Change each gauge weight, check that relative weight changes with each addition
+  const gaugeStartTime = await gaugeController.time_total()
+  for (const gauge of gauges) {
+    const gaugeWeight = (
+      await gaugeController.get_gauge_weight(gauge.address, {
+        gasLimit: 3_000_000,
+      })
+    ).toString()
+    const gaugeRelativeWeight = await gaugeController.callStatic[
+      "gauge_relative_weight_write(address,uint256)"
+    ](gauge.address, gaugeStartTime,{ gasLimit: 3_000_000 })
+
+    console.log(`${gauge.name} gauge_weight: ${gaugeWeight}`)
+    console.log(`${gauge.name} relative_weights: ${gaugeRelativeWeight}`)
+
+    // Imitate multisig setting gauge weights
+    await gaugeController.change_gauge_weight(gauge.address, 10000, {
+      gasLimit: 3_000_000,
+    })
```

Comment:
Would it be possible to move this to the second week launch script before time is advanced?

We expect to use initial gauge weights that were set instead of calling `change_gauge_weight` for the first week.

Reply:
Yeah similar to this PR
https://github.com/saddle-finance/saddle-contract/pull/621